HPCC-14280 Access rights checks should include SERVICE calls

richardkchapman opened the pull request:
This seems to be working ok, but will require either that we sign all plugins
explicitly, or that we make code loaded from plugins to be considered
"trusted" explicitly before it can be used.

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

richardkchapman committed:
HPCC-14280 Access rights checks should include SERVICE calls

This seems to be working ok, but will require either that we sign all plugins
explicitly, or that we make code loaded from plugins to be considered
"trusted" explicitly before it can be used.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman committed:
HPCC-14280 Access rights checks should include SERVICE calls

Code loaded from plugins directory (and the internal system definitions) are
considered implicitly signed.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman commented:
@ghalliday Please review. I have added the implicit signing support.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

ghalliday commented on ecl/hql/hqlexpr.hpp:
```diff
@@ -836,6 +836,7 @@ interface IFileContents : public IInterface
     virtual ISourcePath * querySourcePath() = 0;
     virtual const char * getText() = 0;
     virtual size32_t length() = 0;
+    virtual bool isImplicitlySigned() = 0;
```
minor: could have been a const function.  But then so could some of the other functions in this interface...

richardkchapman replied:
I decided NOT to make it const because the others weren't...
